vortex-array/src/arrays/constant/canonical.rs

```diff
@@ -72,9 +72,12 @@ impl ArrayCanonicalImpl for ConstantArray {
                         .map(|s| ConstantArray::new(s, self.len()).into_array())
                         .collect::<Vec<_>>()
                 });
-                Canonical::Struct(StructArray::try_new(
-                    value.struct_dtype().names().clone(),
+                let Some(struct_dtype) = value.dtype().as_struct() else {
+                    unreachable!()
```

Comment:
Now you can convert these to `value.dtype().as_struct().vortex_expect(...).clone()` Same for the other occurence in struct serde

Reply:
I can but why not add the method to struct array to get back the StructArray

Reply:
This also seems like code duplication

Reply:
Miss read the above